packages/rollup-full-node/src/app/web3-rpc-handler.ts

```diff
@@ -491,21 +491,28 @@ export class DefaultWeb3Handler implements Web3Handler, FullnodeHandler {
         throw Error(msg)
       }
 
-      const receipt: OvmTransactionReceipt = await this.getTransactionReceipt(
-        ovmTxHash,
-        true
-      )
-      log.debug(
-        `Transaction receipt for ${rawOvmTx}: ${JSON.stringify(receipt)}`
-      )
-      if (!receipt || !receipt.status) {
-        log.debug(`Transaction reverted: ${rawOvmTx}, ovmTxHash: ${ovmTxHash}`)
-        throw new RevertError(receipt.revertMessage)
-      }
-
-      await this.processTransactionEvents(receipt)
+      this.context.provider
+        .waitForTransaction(internalTxHash)
+        .then(async () => {
+          const receipt: OvmTransactionReceipt = await this.getTransactionReceipt(
+            ovmTxHash,
+            true
+          )
+          log.debug(
+            `Transaction receipt for ${rawOvmTx}: ${JSON.stringify(receipt)}`
+          )
+          if (!receipt) {
+            log.error(`Unable to find receipt for raw ovm tx: ${rawOvmTx}`)
+            return
```

Comment:
Should never happen, but we should probably throw here since this would be really messed up.
```suggestion
            const msg: string = `Unable to find receipt for raw ovm tx: ${rawOvmTx}`
            log.error(msg)
            throw Error(msg)
```